app/app.go

```diff
@@ -29,12 +29,12 @@ type App struct {
 	GracePeriod      time.Duration
 	ShutdownTimeout  time.Duration
 
-	mainReadinessProbe  Probe
-	mainHealthnessProbe Probe
+	mainReadinessProbe   Probe
+	mainHealthinessProbe Probe
 }
 
 // New returns a new App.
-func New(ctx context.Context, adminPort string) (*App, error) {
+func New(ctx context.Context, adminPort string, adminServerOptions ...ServerOption) (*App, error) {
```

Comment:
I think receiving arguments for constructing a dependency is odd. We should just receive de dependency instead. 

If we create a package for the http.Server (comment bellow) this could be something like:
```
func New(ctx context.Context, adminServer *http.Server) (*App, error) {
   ...
    httpserver.WithMux(mux)(aadminServer) // because we are building the handles of mux inside app.New
   ...
}
```
and be used like 
```
app.New(ctx, httpserver.New(
    httpserver.WithPort(9000),
    httpserver.WithTimeout(t),
)
```

On package `app` we could add a NewDefaultAdminServer() that uses sane default for app.App
```
app.New(ctx, app.NewDefaultAdminServer())
```